crypto/src/memo.rs

```diff
@@ -16,12 +23,107 @@ impl Default for MemoPlaintext {
 }
 
 impl MemoPlaintext {
-    pub fn encrypt(&self, _key: &Address) -> MemoCiphertext {
-        // TODO!
-        let memo_bytes = [0u8; MEMO_CIPHERTEXT_LEN_BYTES];
-        MemoCiphertext(memo_bytes)
+    /// Encrypt a memo, returning its ciphertext.
+    pub fn encrypt(
+        &self,
+        esk: &ka::Secret,
```

Comment:
`esk` is supposed to be used ephemerally, but we need to use it twice, once for the memo encryption and once for the note encryption. I'm not sure if this would be better or worse, but would it make sense to try to join those together, so that the API is harder to misuse?